### PR TITLE
feat: managing teams

### LINKED ISSUE #9208: Managing teams: View all teams in organization
Enable the Organization Admin to see a list of all teams under their organization.

AC:
- [ ] list of teams available for org admin

### PR DESCRIPTION
**Pull request by BartoszJarocki**
# Description

Resolves #9208 #9209 #9210 #9211 #9212 #9213 #9214 #9215 #9216 #9217 #9218

(EDITED by Jordan, PR continues below)

### CODE REVIEW

File: .vscode/launch.json
Comment (jordanh) [inline .vscode/launch.json:33]: pwa-chrome is no longer supported on my version of VS Code, the type is now simply `chrome`

**Comment by jordanh** [comment]
## Description

First off, my sincere apologies for the size of this PR. It passed from @BartoszJarocki to @mwermuth and now I'm taking it over the line.

This PR is a "first cut" of allowing a SuperAdmin to be able to manage their organization. Acceptance Criteria / Test Cases are given below.

## Demo

[If possible, please include a screenshot or gif/video, it'll make it easier for reviewers to understand the scope of the changes and how the change is supposed to work. If you're introducing something new or changing the existing patterns, please share a Loom and explain what decisions you've made and under what circumstances]

## Testing scenarios

First, you must promote the user you're testing with to attain the `ORG_ADMIN` role:

```graphql
query {
  user(email: "your-test-user@dundermifflin.co.uk") {
    id
  }
}
```

```graphql
mutation {
  setOrgUserRole(orgId: "xyzxyzxyz", userId: "local|42424242", role: ORG_ADMIN) {
    ... on SetOrgUserRoleSuccess {
      updatedOrgMember {
        id
        role
      }
    }
  }
}
```

  - [ ] Can see all teams within an Org
  - Navigate to http://localhost:3000/me/organizations
  - Choose your org
  - Click on "Teams" from the sidebar

- [ ] Can create teams
  - From the "Teams" view, click "Add Team"

- [ ] Can add members
   - Choose a Team from the "Teams" view, click "Add Member"

- [ ] Can remove members from any team
   - Choose a Team
   - Click 3-dots menu next to any member -> Remove member

- [ ] Can see all members under my org
   - Navigate to http://localhost:3000/me/organizations
   - Choose your org
   - Click on "Members" from the sidebar

- [ ] Can remove any member from my org
   - Choose 3-dots menu -> Remove member

- [ ] Can promote and demote Org Admin role
   - Choose any other member
   - Promote member to Org Admin
   - Demote member to Org Admin

## Final checklist

- [x] I checked the [code review guidelines](../docs/codeReview.md)
- [x] I have added [Metrics Representative](../docs/codeReview.md#metrics-representative) as reviewer(s) if my PR invovles metrics/data/analytics related changes
- [x] I have performed a self-review of my code, the same way I'd do it for any other team member
- [x] I have tested all cases I listed in the testing scenarios and I haven't found any issues or regressions
- [x] Whenever I took a non-obvious choice I added a comment explaining why I did it this way
- [x] I added the label `One Review Required` if the PR introduces only minor changes, does not contain any architectural changes or does not introduce any new patterns and I think one review is sufficient'
- [x] PR title is human readable and could be used in changelog

File: packages/client/modules/userDashboard/components/OrgUserRow/OrgMemberRow.tsx
Comment (jordanh) [inline packages/client/modules/userDashboard/components/OrgUserRow/OrgMemberRow.tsx:204]: This (and below) was the most significant change I made to what was started:

I created a wholly new control for Org Admin actions vs. actions that can be taken by the BillingLeader. This helped simplify logic when decided what role can take what action

File: packages/client/modules/userDashboard/components/OrgTeamMembers/OrgTeamMembers.tsx
Comment (jordanh) [inline packages/client/modules/userDashboard/components/OrgTeamMembers/OrgTeamMembers.tsx:105]: @nickoferrall I've noted a bug just now that needs resolving before this can be merged. If the `ORG_ADMIN` is not a member of the team, the mutation that creates the MassInvitation link will fail. I can see 4 options:

1. If the ORG_ADMIN clicks the link, the client just-in-time fires a mutation to join this team
2. We do the same thing, but on the server-side: if an ORG_ADMIN tries to create a massInvite token, they are joined to the team first
3. Instead of having the ORG_ADMIN generate the mass-invite token, we masquerade as the Team Lead (ick)
4. We remove the "Add Member" functionality all together

I am split between options #1 and #4 and would love a second opinion. What do you think?

Comment (Dschoordsch) [reply packages/client/modules/userDashboard/components/OrgTeamMembers/OrgTeamMembers.tsx:105]: Why not just allow creating mass invitations from non-team members if they are an org admin? It could be irritating to join all those teams and getting the notifications and all.

File: packages/client/modules/userDashboard/components/OrgTeams/OrgTeams.tsx
Comment (nickoferrall) [inline packages/client/modules/userDashboard/components/OrgTeams/OrgTeams.tsx:57]: +1 if there's only one team, the copy should be "Team" rather than "Teams". We can use the plural helper function for this: https://github.com/ParabolInc/parabol/blob/8c68bf771a350b31c7948e6bd11c5c110e30c977/packages/client/utils/plural.ts

File: packages/client/modules/userDashboard/components/OrgTeamMembers/OrgTeamMembers.tsx
Comment (nickoferrall) [reply packages/client/modules/userDashboard/components/OrgTeamMembers/OrgTeamMembers.tsx:105]: I think that handling everything on the server would be ideal as it's more secure and less error-prone, but I think we should go with option four. 

The PR is already a big 'un. Shipping this without the "Add Member" functionality will still add a lot of value, so I think we should work on "Add Member" in a separate PR.

File: packages/client/modules/userDashboard/components/OrgTeams/OrgTeams.tsx
Comment (nickoferrall) [reply packages/client/modules/userDashboard/components/OrgTeams/OrgTeams.tsx:57]: ^ this is from a review in progress, not complete

**Review by nickoferrall — CHANGES_REQUESTED** [review]

File: packages/client/modules/userDashboard/components/OrgTeamMembers/OrgTeamMembers.tsx
Comment (jordanh) [reply packages/client/modules/userDashboard/components/OrgTeamMembers/OrgTeamMembers.tsx]: @nickoferrall I like your logic there. Ok, I will do that.

> Why not just allow creating mass invitations from non-team members if they are an org admin? It could be irritating to join all those teams and getting the notifications and all.

The underlying data model has the massInvitation belonging to a TeamMember. I suppose the fifth (and best) option would be to modify the massInvitation table to have invitations also be able to belong to non-TeamMembers (i.e. the ORG_ADMIN) as well

I'd probably be nice to get that table off of rethinkdb before making that modification

File: packages/client/components/DeleteTeamDialog.tsx
Comment (jordanh) [reply packages/client/components/DeleteTeamDialog.tsx:40]: I agree! I'll update

File: packages/client/components/DeleteTeamDialogRoot.tsx
Comment (jordanh) [reply packages/client/components/DeleteTeamDialogRoot.tsx:13]: Oh for sure, nice catch

File: packages/client/components/DeleteTeamDialogRoot.tsx
Comment (jordanh) [reply packages/client/components/DeleteTeamDialogRoot.tsx:20]: Thank you, I done did it this way :)

File: packages/client/modules/userDashboard/components/OrgTeamMembers/OrgTeamMembers.tsx
Comment (jordanh) [reply packages/client/modules/userDashboard/components/OrgTeamMembers/OrgTeamMembers.tsx:72]: Nice suggestion, and done!

File: packages/client/modules/userDashboard/components/OrgTeams/OrgTeams.tsx
Comment (jordanh) [reply packages/client/modules/userDashboard/components/OrgTeams/OrgTeams.tsx:54]: Done!

File: packages/client/modules/userDashboard/components/OrgTeams/OrgTeams.tsx
Comment (jordanh) [reply packages/client/modules/userDashboard/components/OrgTeams/OrgTeams.tsx:57]: Done!

File: packages/client/modules/userDashboard/components/OrgUserRow/OrgMemberRow.tsx
Comment (jordanh) [reply packages/client/modules/userDashboard/components/OrgUserRow/OrgMemberRow.tsx:118]: done

File: packages/client/modules/userDashboard/components/OrgTeamMembers/OrgTeamMembersRow.tsx
Comment (jordanh) [reply packages/client/modules/userDashboard/components/OrgTeamMembers/OrgTeamMembersRow.tsx]: Great finds! Fixed up

**Comment by jordanh** [comment]
@nickoferrall thank's for the review! I think I've addressed you're review. If you feel comfortable merging, please do. Else, happy to address anything else or pass to another reviewer.

**Review by nickoferrall — APPROVED** [review]

**Pull request merged**
